btf/btf.go

```diff
@@ -452,7 +496,14 @@ func (sw sliceWriter) Write(p []byte) (int, error) {
 // Returns an error wrapping ErrNotFound if a Type with the given ID
 // does not exist in the Spec.
 func (s *Spec) TypeByID(id TypeID) (Type, error) {
-	return s.types.ByID(id)
+	firstID := s.firstTypeID()
+	i := int(id - firstID)
+
+	if id < firstID || i >= len(s.types) {
```

Comment:
I feel like these should maybe be separate errors? Or, since we know the first and last typeID, we could improve the one here? `expected type ID $id between $firstID and $(firstID + len(s.types))`

Reply:
Changed the error message.